numpy/testing/_private/utils.py

```diff
@@ -706,6 +711,19 @@ def chk_same_position(x_id, y_id, hasval='nan'):
                                 % (hasval), verbose=verbose, header=header,
                                 names=('x', 'y'), precision=precision)
             raise AssertionError(msg)
+        return x_id if x_id.shape else y_id
+
+    def remove_flagged(x, y, flag):
+        """Remove elements in x and y, noting that either may be scalar."""
+        if flag is False:
+            return x, y
+        if flag.all():
+            return array(()), array(())
```

Comment:
nit: I'd use `[]` here for a size-0 array - I'm used to seeing `()` for a 0dim array, so this is a little jarring

Reply:
OK